internal/defact/default.go

```diff
@@ -0,0 +1,262 @@
+		})
+		if err != nil {
+			return exes, errs.Wrap(err, "Error while walking path")
+		}
+	}
+
+	return exes, nil
+}
+
+// UniqueExes filters the array of executables for those that would be selected by the command shell in case of a name collision
+func UniqueExes(exePaths []string, pathext string) ([]string, error) {
+	pathExt := strings.Split(strings.ToLower(pathext), ";")
+	exeFiles := map[string]exeFile{}
+	result := []string{}
+
+	for _, exePath := range exePaths {
+		if rt.GOOS == "windows" {
+			exePath = strings.ToLower(exePath) // Windows is case-insensitive
+		}
+
+		exe := exeFile{exePath, "", ""}
+		ext := filepath.Ext(exePath)
+
+		// We only set the executable extension if PATHEXT is present.
+		// Some macOS builds can contain binaries with periods in their
+		// names and we do not want to strip off suffixes after the period.
+		if funk.Contains(pathExt, ext) {
+			exe.ext = filepath.Ext(exePath)
+		}
+		exe.name = strings.TrimSuffix(filepath.Base(exePath), exe.ext)
+
+		if prevExe, exists := exeFiles[exe.name]; exists {
+			pathsEqual, err := fileutils.PathsEqual(filepath.Dir(exe.fpath), filepath.Dir(prevExe.fpath))
+			if err != nil {
+				return result, errs.Wrap(err, "Could not compare paths")
+			}
+			if !pathsEqual {
+				continue // Earlier PATH entries win
+			}
+			if funk.IndexOf(pathExt, prevExe.ext) < funk.IndexOf(pathExt, exe.ext) {
+				continue // Earlier PATHEXT entries win
+			}
+		}
+
+		exeFiles[exe.name] = exe
+	}
+
+	for _, exe := range exeFiles {
+		result = append(result, exe.fpath)
+	}
+	return result, nil
+}
+
+// shimTargetPath returns the full path to the shim target (adds .bat on Windows)
+func shimTargetPath(targetDir string, path string) string {
+	target := filepath.Clean(filepath.Join(targetDir, filepath.Base(path)))
+	if rt.GOOS != "windows" {
+		return target
+	}
+
+	oldExt := filepath.Ext(target)
+	return target[0:len(target)-len(oldExt)] + ".bat"
+}
+
+func createShimFile(fpath, shimPath string) error {
+	logging.Debug("Shimming %s at %s", fpath, shimPath)
+	exe, err := os.Executable()
+	if err != nil {
+		return errs.Wrap(err, "Could not get State Tool executable")
+	}
+
+	tplParams := map[string]interface{}{
+		"exe":     exe,
+		"command": filepath.Base(fpath),
+		"target":  fpath,
+	}
+	box := packr.NewBox("../../assets/shim")
+	boxFile := "shim.sh"
+	if rt.GOOS == "windows" {
+		boxFile = "shim.bat"
+	}
+	shimBytes := box.Bytes(boxFile)
+	shimStr, err := strutils.ParseTemplate(string(shimBytes), tplParams)
+	if err != nil {
+		return errs.Wrap(err, "Could not parse %s template", boxFile)
+	}
+
+	err = ioutil.WriteFile(shimPath, []byte(shimStr), 0755)
+	if err != nil {
+		return errs.Wrap(err, "failed to write shim command %s", shimPath)
+	}
+	return nil
+}
+
+// createShimFiles creates shims in the target path of all executables found in the bins dir
+// It overwrites existing files, if the overwrite flag is set.
+// On Windows the same executable name can have several file extensions,
```

Comment:
This is true on Mac/Linux too. `.sh` vs no extension.

Reply:
I don't think that you can type `script` and it would run `script.sh` instead.  The file endings in Mac/Linux are solely to give some orientation to the user.